community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_3/runtime/ExecutionContext.scala

```diff
@@ -30,25 +30,29 @@ object ExecutionContext {
 
   def from(x: (String, Any)*): ExecutionContext = apply().newWith(x)
 
-  def apply(m: MutableMap[String, Any] = MutableMaps.empty) = new ExecutionContext(m, 0)
+  def apply(m: MutableMap[String, Any] = MutableMaps.empty) = new ExecutionContext(m, 0, 0)
 
-  def apply(numberOfLongs: Int) = new ExecutionContext(m = null, numberOfLongs = numberOfLongs)
+  def apply(numberOfLongs: Int, numberOfRefs: Int) = new ExecutionContext(m = null, numberOfLongs, numberOfRefs)
 }
 
-case class ExecutionContext private (m: MutableMap[String, Any], numberOfLongs: Int)
+case class ExecutionContext private (m: MutableMap[String, Any], numberOfLongs: Int, numberOfRefs: Int)
   extends MutableMap[String, Any] {
 
   def copyFrom(input: ExecutionContext): Unit = {
-    if (input.numberOfLongs > numberOfLongs)
+    if (input.numberOfLongs > numberOfLongs || input.numberOfRefs > numberOfRefs)
```

Comment:
Is this check necessary in the tight loop?
We will get `java.lang.ArrayIndexOutOfBoundsException` if input is larger.